### ISSUE TITLE
Will training the discriminator model alone help improve my model performance?

### ISSUE THREAD

**Issue by kalai2033** [junyanz/pytorch-CycleGAN-and-pix2pix#1044]
Currently, I'm doing a task of image to image translation with the following dataset using pix2pix model. The task is to enhance the contrast between the text. I have very less such images and its label pairs. I am getting decent results with 300 manually annotated images. But it is not feasible to generate more image pairs. 

Also it is very difficult to create the image pairs. I have created two more different datasets from the above large original image.

**Pixel wise paired dataset**
I have cropped various characters present in the image dataset and created pairs.

**Word pair dataset** 

I trained with all three datasets together. It was worsening the performance compared to model trained with full image dataset.

Is it possible to train only the discriminator with the pixelwise and word wise dataset and train the generator and discriminator with the full image dataset?

What do you think will improve the performance in my case? Thanks in advance.

**Comment by junyanz** [comment]
Training the model on the full image is better, as the generator and discriminator can see its surrounding context. If you only have a few images, you need more data argumentation such as cropping (e.g., '--load_size 512 crop_size 256`) and flipping. You can even add your custom argumentation using a 3rd party [library](https://github.com/aleju/imgaug). If you have your annotation on the cropped patches, maybe you can modify the loss and only calculate the loss on the annotated regions.

**Comment by kalai2033** [comment]
Thanks for your suggestion.

**Comment by junyanz** [comment]
You can multiply your mask in this [line](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/8cda06f7c36b012769efac63adc1a68586b8fb85/models/networks.py#L275).

**Comment by kalai2033** [comment]
Multiply the mask in the sense of multiplying the ground truth with predictions? Is it?

**Comment by junyanz** [comment]
I mean multiplying the loss tensor by the mask.

**Comment by kalai2033** [comment]
Thanks :)
